src/ripple/basics/join.h

```diff
@@ -44,7 +42,7 @@ class CollectionAndDelimiter
     Collection const& collection;
     std::string const delimiter;
 
-    explicit CollectionAndDelimiter(Collection const& c, std::string delim)
+    explicit CollectionAndDelimiter(Collection const& c, std::string&& delim)
```

Comment:
I would probably leave `delim` as a value, not an rvalue. The common use case of passing a cstring will act the same. It will act differently if someone already has a `delim` as a string that they want to pass in. Making this a rvalue might have a performance benefit - it forces the caller to decide if they need to copy the delimiter or if they can just move it. But it also make the interface a little harder to use. For a class like this I'd choose to make the interface easier to use. 

I'm fine with it as-is too. But thought I'd make the note.

Reply:
Yeah, that makes sense. I'll change it back.